libraries/type-length-value/src/state.rs

```diff
@@ -402,7 +595,17 @@ impl<'data> TlvStateMut<'data> {
 
         Ok(&mut self.data[value_start..new_value_end])
     }
+
+    /// Reallocate the given number of bytes for the given SplDiscriminate,
+    /// where no repeating discriminators are allowed
+    pub fn realloc_unique<V: SplDiscriminate>(
```

Comment:
I don't think this has any uniqueness guarantees either, right?
```suggestion
    pub fn realloc_first<V: SplDiscriminate>(
```